meta-lmp-bsp/recipes-bsp/u-boot/u-boot-ostree-scr-fit/imx8mmevk/boot.cmd

```diff
@@ -5,17 +5,141 @@ setenv bootlimit 3
+setenv load_image 'ext4load ${devtype} ${devnum}:2 ${loadaddr} ${image_path}'
+setenv set_blkcnt 'setexpr blkcnt ${filesize} + 0x1ff && setexpr blkcnt ${blkcnt} / 0x200'
+setenv update_image 'run set_blkcnt && mmc dev ${devnum} ${uboot_hwpart} && mmc write ${loadaddr} ${start_blk} ${blkcnt}'
+
+imx_secondary_boot
+
+run bootcmd_otenv
+run bootcmd_getroot
+
+imx_is_closed
+
+# Board is open, so ubootenv is used
+if test -n "${board_is_closed}"; then
+	echo "Using FIOVB RPMB"
+else
+	echo "Using ubootenv"
+	setenv fiovb.bootcount "${bootcount}"
+	setenv fiovb.rollback "${rollback}"
+	setenv fiovb.upgrade_available "${upgrade_available}"
+	setenv fiovb.bootupgrade_available "${bootupgrade_available}"
+fi
+
+if test "${fiovb.is_secondary_boot}" = "0"; then
+	if test -n "${board_is_closed}"; then
+		# FIO bootcount, upgrade_available and rollback initialization
+		mmc rescan
+		fiovb init {devnum}
+
+		fiovb read_pvalue bootcount 4
+		if test ! $? -eq 0; then fiovb write_pvalue bootcount 0; fi
+
+		fiovb read_pvalue rollback 4
+		if test ! $? -eq 0; then fiovb write_pvalue rollback 0; fi
+
+		fiovb read_pvalue upgrade_available 4
+		if test ! $? -eq 0; then fiovb write_pvalue upgrade_available 0; fi
+
+		fiovb read_pvalue bootupgrade_available 4
+		if test ! $? -eq 0; then fiovb write_pvalue bootupgrade_available 0; fi
+	fi
+
+	# Only update bootcount when upgrade_available is set
+	if test "${fiovb.upgrade_available}" = "1"; then
+		setexpr bootcount ${fiovb.bootcount} + 1;
+		if test -n "${board_is_closed}"; then
+			fiovb write_pvalue bootcount ${bootcount};
+		fi;
+
+		# If bootcount is above bootlimit, trigger rollback
+		# if not - try to boot secondary boot path for boot firmware validation
+		if test ${bootcount} -gt ${bootlimit}; then
+			echo "FIO: doing rollback..."
+			if test -n "${board_is_closed}"; then
+				fiovb write_pvalue rollback 1;
+				fiovb write_pvalue upgrade_available 0;
+				fiovb write_pvalue bootupgrade_available 0;
+			else
+				setenv rollback 1;
+				setenv upgrade_available 0;
+				setenv bootupgrade_available 0;
+			fi;
+		else
+			if test "${fiovb.bootupgrade_available}" = "1"; then
+				imx_secondary_boot 1;
+				echo "FIO: updating secondary boot images from ${ostree_root} ..."
+
+				setenv image_path "${ostree_root}/usr/lib/firmware/imx-boot"
+				echo "FIO: Writing ${image_path} ..."
+				setenv start_blk 1042
+				run load_image
+				run update_image
+
+				setenv image_path "${ostree_root}/usr/lib/firmware/u-boot.itb"
+				echo "FIO: Writing ${image_path} ..."
+				setenv start_blk 1300
+				run load_image
+				run update_image
+
+				run saveenv_mmc
+				echo "FIO: doing warm reset to boot into secondary boot path..."
+				imx_warm_reset
+			fi;
+		fi
+	fi
+
+	# That means that we've validated already boot image over secondary path
+	if test "${fiovb.bootupgrade_available}" = "1" && test "${fiovb.upgrade_available}" = "0"; then
+		echo "FIO: update primary boot path path with validated images ..."
```

Comment:
I think "path" is duplicated here, could you double check?

Reply:
yes, will fix, thanks